include/Gaffer/MetadataAlgo.h

```diff
@@ -135,6 +138,54 @@ GAFFER_API Node *getNumericBookmark( ScriptNode *script, int bookmark );
 GAFFER_API int numericBookmark( const Node *node );
 GAFFER_API bool numericBookmarkAffectedByChange( const IECore::InternedString &changedKey );
 
+/// Annotations
+/// ===========
+///
+/// Annotations define arbitrary text to be displayed in a coloured area
+/// next to a node. Each node can have arbitrary numbers of annotations,
+/// with different annotations being distinguished by their `name`.
+/// Templates can be used to define defaults for standard annotation types.
```

Comment:
It took me some searching to figure out the template is just for setting color, as opposed to using some sort of substitution based tokens in a template string.  I guess you're anticipating more complex uses later, but it might help to give an example of what this currently means.